rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -3382,6 +3384,226 @@ int LuaUnsyncedCtrl::SetBuildFacing(lua_State* L)
 }
 
 
+static bool ParseBuildInfo(
+	lua_State* L,
+	BuildInfo& buildInfo,
+	const char* caller,
+	int idx
+) {
+	for (int i=0; i<5; i++) {
+		if (!lua_isnumber(L, idx+i)) {
+			luaL_error(L, "%s(): bad build info type", caller);
+			return false;
+		}
+	}
+
```

Comment:
Use `luaL_checkinteger` (or float etc. as appropriate) which has a built-in type check, then get rid of all that explicit type check bloat.

Reply:
done